res/res_rtp_asterisk.c

```diff
@@ -1914,6 +1913,27 @@ struct dtls_cert_info {
 	X509 *certificate;
 };
 
+static int apply_dh_params(SSL_CTX *ctx, BIO *bio)
+{
+	int res = 0;
+
+#if OPENSSL_VERSION_NUMBER >= 0x30000000L
+	EVP_PKEY *dhpkey = PEM_read_bio_Parameters(bio, NULL);
+	if (dhpkey) {
+		res = EVP_PKEY_is_a(dhpkey, "DH") && SSL_CTX_set0_tmp_dh_pkey(ctx, dhpkey);
+	}
+	EVP_PKEY_free(dhpkey);
```

Comment:
EVP_PKEY_free should be inside if (dhpkey) {}

Reply:
[From `man 3 EVP_PKEY_free`](https://www.openssl.org/docs/man3.0/man3/EVP_PKEY_free.html):

> EVP_PKEY_free() decrements the reference count of key and, if the reference count is zero, frees it up. If key is NULL, nothing is done.